### ISSUE TITLE
~0% accuracy on counting training data, 42.5% accuracy on evaluation data

### ISSUE THREAD

**Issue by ZhipengCai** [Deep-Agent/R1-V#155]
Thanks for the amazing work! I have an issue reproducing grpo training for the counting task on [🤗 R1V Training Dataset: CLEVR-70k-Counting](https://huggingface.co/datasets/leonardPKU/clevr_cogen_a_train)

According to your report, the evaluation accuracy on SuperCLEVR is 43% with QWen2-VL, which is similar on my side. But for the accuracy reward during training, initially your curve shows ~50% accuracy while I only got ~5%. And if I run the same evaluation code on the training dataset, i.e., CLEVR-70k-Counting, the accuracy is nearly 0%. Do you have any idea on why this is the case, and how you achieve ~50% training accuracy at the very beginning?

My training code should be fine since I mostly reused your code and the format reward is going up. I used non-vllm grpo trainer, with the same code and script you share. I dont think the issue lies in training since I run the evaluation code on training data and the accuracy is already ~0%, and the same evaluation code can reproduce the accuracy on test data. Several things I noticed: 

1) you used different system prompts during training and evaluation.
2) SuperCLEVR and ClEVR-70k used different question prompts (How many different items vs How many items).
3) The input images are with different sizes.
4) Due to some issues, I can only use transformers 4.46, but this is the same for evaluation on both dataset.

Do you have any idea on how I can reproduce your results? Thanks!

Thanks

**Comment by Alyroy** [comment]
i also have the same questoin, may i ask for an anwser if you get it?
